content/post/000-ipfs-in-web-browsers.md

```diff
@@ -0,0 +1,74 @@
+
+(INSERT STAGES DIAGRAM)
+
+## Firefox
+
+> Perhaps it makes sense to mention Firefox as a full recap-like section here, afaik we did not blog about libdweb at all yet. 
+> Idea: start with mentioning we've been working with Mozilla on initial APIs since 2018, mention [experimental protocol handler](https://github.com/ipfs-shipyard/ipfs-companion/pull/533) and [local discovery + TCP transport](https://github.com/ipfs-shipyard/ipfs-companion/pull/553) and include video demos from 2018: user-facing handler https://www.youtube.com/watch?v=fS8pLOQdOoM + optionally https://www.youtube.com/watch?v=FRzyWUXIyeo then end this section with hopes to see those APIs start landing in Nightly as a soft pressure by shining some light on this. 
+> Thoughts? [name=lidel]
+
+## Brave
+
+Brave has tended to live up to its name since the beginning - riling up the browser, publisher and crypto worlds. [Since 2018](https://github.com/brave/brave-browser/issues/819) we've been working with Brave on IPFS support in some form, and the results of that work are now emerging.
+
+Today you can open [brave://settings/extensions](brave://settings/extensions) in Brave beta or nightly and see a toggle for enabling IPFS Companion.
+
+![Enabling IPFS Companion in Brave](https://file.globalupload.io/ZVxjmPZ48E.gif)
+
+Removing this install step *really matters* for adoption. While technical users are familiar with browser extensions, it's less comfortable for non-technical users.
+
+In the future, when you [load a URL beginning with "ipfs://"](https://github.com/brave/brave-browser/issues/3045), Brave will prompt you to enable the extension, providing a smooth discovery and onboarding experience.
+
+But that isn't all. One of the biggest barriers to even just experimenting with true p2p networks in browsers is the lack of APIs available to extensions and web pages. The primitives of most p2p systems are filesystem access, raw TCP sockets, UDP sockets for broadcast and receive, and in browsers you need a way to navigate using a custom protocol. We've figured out one piece of that puzzle while working with Brave: Chromium OS sockets!
+
+[Chromium OS](https://en.wikipedia.org/wiki/Chromium_OS) has a raw sockets API, but it's disabled in the regular browser. Brave has modified their build to whitelist those APIs for IPFS Companion to use. 
+
+These powerful APIs enable the js-ipfs node embedded in the browser to provide a *true* P2P experience without the need for an external daemon:
+
+* Embedded HTTP Gateway removes reliance on public gateways
+* TCP transport improves connectivity (enables direct interop with go-ipfs)
```

Comment:
by "direct" - do you mean "using the IPFS protocol instead of going through http" or something else? might not be clear to regular companion users that they don't have a "direct connection" to their local go-ipfs node...

Reply:
fixed